src/resource/solidDataset.ts

```diff
@@ -705,7 +705,9 @@ export async function saveSolidDatasetInContainer(
  * will do the job just fine, and does not require the parent Container to exist in advance.
  *
  * @param containerUrl URL of the Container in which the empty Container is to be created.
- * @param options Optional parameter `options.fetch`: An alternative `fetch` function to make the HTTP request, compatible with the browser-native [fetch API](https://developer.mozilla.org/docs/Web/API/WindowOrWorkerGlobalScope/fetch#parameters).
+ * @param options Optional parameter `options.fetch`: An alternative `fetch` function to make the HTTP request, compatible with the browser-native [fetch API](https://developer.mozilla.org/docs/Web/API/WindowOrWorkerGlobalScope/fetch#parameters).`options.slugSuggestion` accepts a string for your new Container's name.
+ * @returns A promise that resolves to a SolidDataset if successful, and that
```

Comment:
just for my own clarification: so if it returns as "rejected" it won't be because you tried to `createContainerInContainer` where the parent container was a local or unpersisted because according to the comment thats okay? so you will only get a reject if the container fails to create for some reason?

Just

Reply:
Typically it'll be because you don't have permission or the parent container doesn't exist, or the container you're trying to create already exists, etc — but "slugSuggestion" is exactly that: you may call this and get a container back that isn't what your slugSuggestion is.

Reply:
See: https://solidproject.org/TR/protocol#resource-type-heuristics

Reply:
Comment addressed in newest commit!